### PR TITLE
BAAS-27268: add early exit in mem usage where missing

### PR DESCRIPTION
**Pull request by Gabri3l**
I deleted two tests that were redundant and were cumbersome to figure out the proper mem limit number. Even for future changes I think it would have been an unnecessary complication.

### CODE REVIEW

File: runtime.go [hunk 1]
```diff
@@ -515,6 +515,9 @@ func (r *Runtime) MemUsage(ctx *MemUsageContext) (memUsage uint64, newMemUsage u
 			if err != nil {
 				return memUsage, newMemUsage, err
 			}
+			if exceeded := ctx.MemUsageLimitExceeded(memUsage); exceeded {
```
Comment (LijieZhang1998) [inline runtime.go:521]: [q] Do we need to add the same check after Line 503 for `r.globalObject`?

**Review by LijieZhang1998 — COMMENTED** [review]
Nice to have memory usage improvement!! I have one question.

File: object_gomap_test.go [hunk 2]
```diff
@@ -430,17 +435,40 @@ func TestGoMapMemUsage(t *testing.T) {
...
+				data: map[string]interface{}{
+					"test0": valueInt(99),
+					"test1": valueInt(99),
+					"test2": valueInt(99),
+					"test3": valueInt(99),
+					"test4": valueInt(99),
+					"test5": valueInt(99),
+				},
+			},
+			memLimit: 0,
```
Comment (nickpoindexter) [inline object_gomap_test.go:460]: [q] Does zero work for this edge case? We don't just exit early. I would think there are cases where we check for memlimit zero.

**Review by nickpoindexter — APPROVED** [review]
LGTM! Just making sure the zero mem limit tests are valid.

File: object_gomap_test.go (hunk 2 above)
Comment (Gabri3l) [reply object_gomap_test.go:460]: It works as expected here, pretty much as soon as we enter the loop we are exceeding the limit, I can increase this to include 2 values but it doesn't change much. You can see the expected memory accounts for at least one entry to be accounted for.

File: runtime.go (hunk 1 above)
Comment (Gabri3l) [reply runtime.go:521]: I definitely can, it might not be as effective but it doesn't hurt, let me make the change!

File: object_gomap_test.go (hunk 2 above)
Comment (nickpoindexter) [reply object_gomap_test.go:460]: 🖖 Sounds good just wanted to make sure there weren't any potential `== 0` checks that might make this test void.

**Review by LijieZhang1998 — APPROVED** [review]
